### PR TITLE
PIA-1715: Gateway traffic going through the tunnel

### PR DESCRIPTION
**Pull request by kp-iva-nedeleva**

### CODE REVIEW

File: vpnmanager/vpnservicemanager/src/main/java/com/kape/vpnservicemanager/data/externals/Service.kt [hunk 1]
```diff
@@ -41,6 +41,7 @@ internal class Service :
     private lateinit var protocol: IProtocol
     private lateinit var subnet: ISubnet
     private lateinit var cache: ICache
+    private var gateway: String? = null
```
Comment (kp-juan-docal) [inline vpnmanager/vpnservicemanager/src/main/java/com/kape/vpnservicemanager/data/externals/Service.kt:44]: use the `cache`. You create the set/get/clearGateway and make sure to add the clear as part of the `clear()` method you'll find there. In general we don't want to add instance variables on the objects but rather define a cache interface (if needed) for what we want to do, and let the cache deal with states.

File: vpnmanager/vpnservicemanager/src/main/java/com/kape/vpnservicemanager/data/externals/Service.kt [hunk 2]
```diff
@@ -82,6 +83,7 @@ internal class Service :
         protocol.stopConnection(disconnectReason).getOrElse {
             return Result.failure(it)
         }
+        gateway = null
```
Comment (kp-juan-docal) [inline vpnmanager/vpnservicemanager/src/main/java/com/kape/vpnservicemanager/data/externals/Service.kt:86]: this is not needed once we move it to `cache`.

File: build.gradle [hunk 3]
```diff
@@ -12,7 +12,7 @@ plugins {
 
 allprojects {
     group = "com.kape.android"
-    version = "0.3.5"
+    version = "0.3.6"
```
Comment (kp-juan-docal) [inline build.gradle:15]: let's make the bump on a separate PR.

**Review by kp-juan-docal — COMMENTED** [review]
The reason why wireguard is not working is probably because we are not setting the `serverVip` [here](https://github.com/pia-foss/mobile-android-vpn-manager/blob/main/vpnmanager/vpnservicemanager/vpnprotocol/src/main/java/com/kape/vpnprotocol/domain/usecases/wireguard/GenerateWireguardSettings.kt#L61) when LAN is enabled, we are only doing it on the service side. The info to do it is there I believe, the configuration to check the LAN state, and the serverVip.

File: vpnmanager/vpnservicemanager/src/main/java/com/kape/vpnservicemanager/data/externals/Service.kt (hunk 1 above)
Comment (kp-iva-nedeleva) [reply vpnmanager/vpnservicemanager/src/main/java/com/kape/vpnservicemanager/data/externals/Service.kt:44]: Sure thing, will do.

File: vpnmanager/vpnservicemanager/src/main/java/com/kape/vpnservicemanager/data/externals/Service.kt (hunk 2 above)
Comment (kp-iva-nedeleva) [reply vpnmanager/vpnservicemanager/src/main/java/com/kape/vpnservicemanager/data/externals/Service.kt]: addressed.

File: build.gradle (hunk 3 above)
Comment (kp-iva-nedeleva) [reply build.gradle]: reverted the change

**Review by kp-juan-docal — APPROVED** [review]

**Review by kp-juan-docal — APPROVED** [review]
